### PR TITLE
integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Ibizugbe**
In this PR, I have completed the current milestone of the project and kept to the given requirements.
- the project is professionally documented.
- the right git-flow method has been used.
- no linter error.
- The n+1 problem has been solved when fetching posts and their comments for a user.
- capybara has been used to write tests for the user show page, user index page,  post show page, and post index page.

### CODE REVIEW

File: spec/views/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    expect(page).to have_content(@user2.name)
+  end
+
+  scenario 'displays all users profile photo' do
+    visit user_path(id: @user.id)
+    visit user_path(id: @user2.id)
+    find("img[src='https://unsplash.com/photos/F_-0BxGuVvo']")
+    find("img[src='https://unsplash.com/photos/F_-0BxGuVvo']")
+  end
+
```
Comment (mikethreels) [inline spec/views/user_index_spec.rb:22]: - [ ] for the user index page you should also write a test which Tests if redirects to the user's show page when you click on a user works.

File: spec/views/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  it 'shows number of user posts ' do
+    expect(page).to have_content('Number of Posts : 0')
+  end
+
+  it 'shows post comments' do
+    expect(page).to have_content('I am a fashion designer with 2 decades of experience')
+  end
+end
```
Comment (mikethreels) [inline spec/views/user_show_spec.rb]: - [ ] for the user show page you should write the below tests as well:
  - Tests that asserts you can see the user's first 3 posts.
  - Tests that asserts you can see a button that lets me view all of a user's posts.
  - Tests that redirects to the post's show page when you click on a user's post.
  - Tests that redirects to the user's post's index page when you click on a see all posts.

File: spec/views/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    expect(page).to have_content('Hello')
+  end
+
+  scenario 'display the how many comments' do
+    expect(page).to have_content('Comments: 5')
+  end
+
+  scenario 'display the how many Likes' do
+    expect(page).to have_content('Likes: 0')
+  end
```
Comment (mikethreels) [inline spec/views/post_index_spec.rb]: - [ ] for the post index page you are expected to write the following tests as well:
  - Tests that asserts you can see the user's profile picture.
  - Tests that asserts you can see the user's username.
  - Tests that asserts you can see the number of posts the user has written.
  - Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
  - Tests that redirects to the post's show page when you click on a post.

File: spec/views/post_show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    expect(page).to have_content('Comments: 5')
+  end
+
+  scenario 'displays number of likes' do
+    expect(page).to have_content('Likes: 0')
+  end
+
+  scenario 'displays the content of the post' do
+    expect(page).to have_content(@first_post.text)
+  end
```
Comment (mikethreels) [inline spec/views/post_show_spec.rb]: - [ ] for the post show page you are expected to include the following tests as well:
  - Tests that asserts you can see who wrote the post.
  - Tests that asserts you can see the username of each commenter. 
  - Tests that asserts you can see the comment each commentor left.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Abraham, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._
### Because you didn't make changes in the post controller I'm leaving one of the requirements in this comment
- [ ] Your app should solve the N+1 problem when fetching all posts and their comments for a user as shown in the [wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_all_posts.png?token=ABIFLAZ4K2ZHA777536NECDBHORQS)
this means that the below part of code will look something like `@posts = @user.posts.includes(:comments)`
https://github.com/Ibizugbe/BlogOnRails/blob/7143e968f524fc7860ac25eab19329c2653d6885/app/controllers/posts_controller.rb#L2-L5



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ibizugbe** [comment]
Thanks @mikethreels  for reviewing our project, we have implemented the requested changes.

File: spec/views/post_show_spec.rb (hunk 4 above)
Comment (Ibizugbe) [reply spec/views/post_show_spec.rb]: Thanks for this observation, we have fixed it.

File: spec/views/post_index_spec.rb (hunk 3 above)
Comment (Ibizugbe) [reply spec/views/post_index_spec.rb]: Thanks for pointing these out, we will make adjustments immediately.

**Review by ShadyShawkat — APPROVED** [review]
## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All tests are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ibizugbe** [comment]
Thanks, @ShadyShawkat for reviewing and approving this PR, I really appreciate🙏

**Pull request merged**